### PR TITLE
Remove migration sticker

### LINKED ISSUE #92032: Migration Flow Improvements: Clean up the migration flag when the user quits the migration flow
### Details

When the user enters the migration flow we set a corresponding flag: https://github.com/Automattic/wp-calypso/issues/91788

We need to clean it up when they successfully complete the purchase or exit the flow.

Scenarios to consider:
- Clean up the the flag on a click to the back button.
- Clean up the flag before choosing migrate/import or when we enter import branch.
- Clean up the the flag after checkout.

Other scenarios to consider?

We want to have a [feature flag](https://github.com/Automattic/wp-calypso/pull/92147/commits/ed8650d7a4f9ab497efb81d91d30d43763091c29) to disable this behavior.

### Checklist

_No response_

### Related

https://github.com/Automattic/dotcom-forge/issues/7759

### PR DESCRIPTION
**Pull request by Imran92**
Related to https://github.com/Automattic/wp-calypso/issues/92032

## Proposed Changes

Removes the migration sticker from the site when the user completes checkout or goes out of the flow. We've covered the usual flows-
- Unmounting/Navigating back from the "Upgrade" page
- Completing the checkout
- Clicking the button for importing the content only

## Why are these changes being made?

* To remove the migration discount sticker from the site

## Testing Instructions

- Enable the feature flag `migration-flow/introductory-offer`. You can add it to your `.env` file like this `ACTIVE_FEATURE_FLAGS="migration-flow/new-migration-instructions-step,migration-flow/introductory-offer"` 
- Start monitoring your browser requests.
- Start a migration and navigate until the step `site-migration-upgrade-plan`.
- Check that when that step loads, it requests the endpoint _.../{SITE_ID}/migration-flow_
- Now click the Back button on top left
- Make sure you see another request sent to the same URL as before, but it's a DELETE request
- Go back to the Migrate/Import page and click on the button for importing content only
- Make sure you see the Delete request being sent
- Now go to the upgrade page, select Upgrade and Migrate, and complete the checkout
- Make sure you see this call being made on the next page

## Pre-merge Checklist

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [x] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/blocks/importer/wordpress/upgrade-plan/with-migration-sticker.tsx
Comment (renatho) [inline client/blocks/importer/wordpress/upgrade-plan/with-migration-sticker.tsx:30]: Just a double-check with @merkushin, we won't need the sticker in the checkout step for anything, right?

**Review by renatho — APPROVED** [review]
Looks good and works well!

I'd just wait for the source branch to be merged before merging this, so the review doesn't get complicated with 2 PRs in one. =)

Also, I just added a comment for a double-check with Dmitry, just in case he explored something about the checkout that we don't know. 😉

File: client/blocks/importer/wordpress/upgrade-plan/with-migration-sticker.tsx
Comment (merkushin) [reply client/blocks/importer/wordpress/upgrade-plan/with-migration-sticker.tsx:30]: We need it until the end of the checkout. We can remove it when the plan was paid successfully.

**Comment by merkushin** [comment]
Hi @Imran92! There are a lot of conflicts it the PR. Could you solve them?

But I need to ask you to rebase your branch on `add/migration-flow-display-intro-offer` before that. In that branch, we already solved some conflicts and moved the wrapping from the UpgradePlanDetails component to UpgradePlan component (that's because fetching data moved to it). Maybe it is worth waiting for merging this [PR](https://github.com/Automattic/wp-calypso/pull/92201) first. 

Honestly, I'm not sure where it will be easier to solve the conflicts, but that PR is in priority for the project.

**Comment by Imran92** [comment]
Thanks @merkushin, I've fixed all the conflicts. There were way too many conflicts, many of those were in files this PR doesn't event touch :p I wonder what happened that caused these many issues!

File: client/blocks/importer/wordpress/upgrade-plan/with-migration-sticker.tsx
Comment (Imran92) [reply client/blocks/importer/wordpress/upgrade-plan/with-migration-sticker.tsx:30]: The "Upgrade" step does not get unmounted when `Checkout` opens as it opens in a modal. So the sticker won't get unset when we open the checkout.

Comment (renatho) [reply client/blocks/importer/wordpress/upgrade-plan/with-migration-sticker.tsx:30]: TIL! I thought it was a different step! Thank you for the answer! =)

**Review by renatho — APPROVED** [review]
Approving again! Thank you for the answer to my question!

**Comment by merkushin** [comment]
@Imran92 I merged the PR for displaying the intro offer, that resulted in a new bunch of conflicts here.

> There were way too many conflicts, many of those were in files this PR doesn't event touch :p I wonder what happened that caused these many issues!

I think that happens because we always squash commits while merging. I just got a bunch of conflicts in another PR that didn't touch any JS, but the "original" branch also had some changes after branching out.

**Comment by Imran92** [comment]
Fixed the conflicts again, as it was already approved, I'll go ahead and merge this to prevent more conflict issues

**Review by merkushin — APPROVED** [review]

**Pull request merged**
